baseapp/baseapp.go

```diff
@@ -1007,6 +1007,11 @@ func (app *BaseApp) runTx(ctx sdk.Context, mode runTxMode, tx sdk.Tx, checksum [
 	if ctx.CheckTxCallback() != nil {
 		ctx.CheckTxCallback()(ctx, err)
 	}
+	resultStr := ""
+	res := app.getDeliverTxResponse(ctx, gInfo, result, &resultStr)
+	for _, hook := range app.deliverTxHooks {
+		hook(ctx, tx, checksum, res)
```

Comment:
this will affect the actual gas Used for the tx execution, but will be using the `res` with a stale gas used from getDeliverTxResponse. is there a scenario where this difference between the deliverTx response used for hooks and deliver tx response returned via tendermint causes an issue?

Reply:
hooks (as of right now) don't use gasUsed passed in. I changed the data structure to be one specific to delivertx hooks that exclude gasUsed to avoid future misuse.